### PR TITLE
Deprecate the BigInteger enum `Round` in favor of a new enum `round`

### LINKED ISSUE #17723: [Library Stabilization] Capitalization of Round enum constants in BigInteger module
The `Round` enum uses all caps for the constants.  Our standard module best practices guide says "Enum type names should be camelCase. The enum values should also be camelCase."  This enum should be updated to follow that guideline

### PR DESCRIPTION
**Pull request by lydia-duncan**
This new enum also uses all lowercase for the constants instead of all caps.

This meant deprecating the functions that took `Round` as an argument in favor
of new versions that take `round`.  This impacted:
- two versions of div_q
- two versions of div_r
- two versions of div_qr
- div_q_2exp
- div_r_2exp

When deprecating, replace the default argument value with just a type
declaration to avoid potential conflicts when relying on the default value (such
calls will naturally switch to using the new type)

Resolves #17723 

Replace uses of `Round` enum and its constants in tests with new `round` enum.

Adds a test of the deprecation warning for all these symbols

Passed a full paratest with futures

### CODE REVIEW

**Comment by lydia-duncan** [comment]
This PR should be merged after #17839, as it will break a few tests otherwise.

File: modules/standard/BigInteger.chpl [hunk 1]
```diff
@@ -4044,38 +4066,69 @@ When ``n/d`` does not produce an integer, this method may produce incorrect resu
...
 
     this.div_q(n, new bigint(d), rounding);
   }
 
+  deprecated
+  "bigint.div_r using Round is deprecated, use bigint.div_r with round instead"
   proc bigint.div_r(const ref n: bigint,
                     const ref d: bigint,
-                    param     rounding = Round.ZERO) {
+                    param     rounding: Round) {
```
Comment (Maxrimus) [inline modules/standard/BigInteger.chpl:4103]: Why was this changed from Round.ZERO to Round?

**Review by dlongnecke-cray — APPROVED** [review]
Awesome to see the new deprecate feature at work! LGTM.

File: modules/standard/BigInteger.chpl (hunk 1 above)
Comment (dlongnecke-cray) [reply modules/standard/BigInteger.chpl:4103]: See OP:

> When deprecating, replace the default argument value with just a type
declaration to avoid potential conflicts when relying on the default value (such
calls will naturally switch to using the new type)

**Review by Maxrimus — APPROVED** [review]

**Comment by lydia-duncan** [comment]
Thanks :)
